infra/azure/terraform/capz/variables.tf

```diff
@@ -31,3 +31,9 @@ variable "storage_account_name" {
   default     = "k8sprowstoragecomm"
   description = "Name of the storage account."
 }
+
+variable "subscription_id" {
+  type        = string
+  default     = "46678f10-4bbb-447e-98e8-d2829589f2d8"
+  description = "Azure Subscription ID to use for the azurerm provider."

```

Comment:
not sure why you added this specifically, but I just a note that I had to still switch my `az cli` login to the correct sub to run `terraform init`.  But this does make it clear that this is the sub we wish to be operating on in the rest of the scripts.

Reply:
I was having issues where terraform wasn't picking up the subscription from my az cli login context. 
I don't remember what exactly the error was but this fixed it.